backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java

```diff
@@ -0,0 +1,89 @@
+package gov.cdc.usds.simplereport.service;
+
+import feign.FeignException;
+import gov.cdc.usds.simplereport.api.model.errors.CsvProcessingException;
+import gov.cdc.usds.simplereport.api.model.errors.IllegalGraphqlArgumentException;
+import gov.cdc.usds.simplereport.config.AuthorizationConfiguration;
+import gov.cdc.usds.simplereport.db.model.Organization;
+import gov.cdc.usds.simplereport.db.model.TestResultUpload;
+import gov.cdc.usds.simplereport.db.model.auxiliary.UploadStatus;
+import gov.cdc.usds.simplereport.db.repository.TestResultUploadRepository;
+import gov.cdc.usds.simplereport.service.model.reportstream.FeedbackMessage;
+import gov.cdc.usds.simplereport.service.model.reportstream.ReportStreamStatus;
+import gov.cdc.usds.simplereport.service.model.reportstream.UploadResponse;
+import java.io.IOException;
+import java.io.InputStream;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class TestResultUploadService {
+  private final TestResultUploadRepository _repo;
+  private final DataHubClient _client;
+  private final OrganizationService _orgService;
+
+  @AuthorizationConfiguration.RequirePermissionCSVUpload
+  public TestResultUpload processResultCSV(InputStream csvStream)
+      throws IllegalGraphqlArgumentException {
+
+    TestResultUpload result = new TestResultUpload(UploadStatus.FAILURE);
+
+    Organization org = _orgService.getCurrentOrganization();
+
+    byte[] content;
+    try {
+      content = csvStream.readAllBytes();
+    } catch (IOException e) {
+      log.error("Error reading test result upload CSV", e);
+      throw new CsvProcessingException("Unable to read csv");
+    }
+
+    UploadResponse response = null;
+    if (content.length > 0) {
+      try {
+        response = _client.uploadCSV(content);
+      } catch (FeignException.BadRequest e) {
+        log.warn("CSV upload bad request", e);
+        result.setErrors(new FeedbackMessage[] {new FeedbackMessage("api", "Bad Request")});
+      } catch (FeignException fe) {
+        log.error("Error connecting to Report Stream", fe);
+        result.setErrors(new FeedbackMessage[] {new FeedbackMessage("api", "Server Error")});
```

Comment:
idk how I feel with recording completely bad requests. we shouldn't be storing anything unless we actually make the call to report stream and get a response from them, otherwise whats the point, open to discuss

Reply:
Anything that lands in these catch clauses shouldn't cause anything to get written to the DB if that's what you're getting at (its looking at the response object not the result).